bpf/lib/nat.h

```diff
@@ -222,17 +212,14 @@ static __always_inline int snat_v4_new_mapping(struct __ctx_buff *ctx, void *map
 	ret = DROP_NAT_NO_MAPPING;
 	goto out;
 
-create_nat_entries:
+create_nat_entry:
 	ostate->to_sport = rtuple.dport;
-	ostate->common.created = bpf_mono_now();
-	rstate.common.created = ostate->common.created;
+	ostate->common.created = rstate.common.created;
 
-	/* Create the SNAT and RevSNAT entries. We just confirmed that
-	 * this RevSNAT entry doesn't exist yet, and the caller previously
-	 * checked that no SNAT entry for this connection exists.
-	 */
-	ret = __snat_update(map, otuple, ostate, &rtuple, &rstate);
+	/* Create the SNAT entry. We just created the RevSNAT entry. */
+	ret = __snat_create(map, otuple, ostate);
 	if (ret < 0) {
+		map_delete_elem(map, &rtuple); /* rollback */
 		if (ext_err)
```

Comment:
fwiw, I believe we only reach  `snat_v4_new_mapping()` if the preceding SNAT logic didn't find any SNAT entry for the connection. So at least an `-EEXIST` here would be *very* unexpected.

Reply:
Agreed. Do you want me to do something about this?

Reply:
I wouldn't touch it as part of this PR, no. Preserving the old behaviour for now makes sense. And in the end we couldn't delete the rollback path - I suspect we would just want to install the SNAT entry without `BPF_NOEXIST`, but still handly unexpected errors.